.circleci/config.yml

```diff
@@ -45,6 +48,16 @@ jobs:
             mkdir /tmp/artifacts && cp version.json /tmp/artifacts/version.js
 
       - setup_remote_docker
+
+      - run:
+          run: Login to Dockerhub
+          command: |
+            if [ "${DOCKER_USER}" == "" ] || [ "${DOCKER_PASS}" == "" ]; then
+              echo "Skipping login to Dockerhub, credentials not available."
+            else
+              echo "${DOCKER_PASS}" | docker login -u="${DOCKER_USER}" --password-stdin
```

Comment:
Is the a particular reason why we're piping the output from `echo` into `docker login`, rather than using `-p`?

Reply:
Good question, I was following what was recommended in [this post mentioned by `ckolos`](https://mozilla.slack.com/archives/G4F4LN4TF/p1602108922071900).

Also, docker complains if you `-p`:
```
WARNING! Using --password via the CLI is insecure. Use --password-stdin.
```

Reply:
> Good question, I was following what was recommended in [this post mentioned by `ckolos`](https://mozilla.slack.com/archives/G4F4LN4TF/p1602108922071900).

Was it actually recommended, or just the way it was present in that diff? I might be missing something -- can't see an explanation or recommendation in the link you posted about that particular implementation.

> Also, docker complains if you `-p`:
> 
> ```
> WARNING! Using --password via the CLI is insecure. Use --password-stdin.
> ```

I imagine it is insecure in interactive mode, but if it's in non-interactive mode, I can't see what difference it would make since the password will still be stored in history even when passed to `echo`. 

As per Docker's documentation, this only provides some level of security if you store the password in a file (and not pass it through `echo`: https://docs.docker.com/engine/reference/commandline/login/#provide-a-password-using-stdin.

Not a blocker, but seems a little redundant/unnecessary, and perhaps gives a false sense of security.

Reply:
> Was it actually recommended, or just the way it was present in that diff?

It wasn't explicitly recommended, it was just present in the diff.

> I imagine it is insecure in interactive mode, but if it's in non-interactive mode, I can't see what difference it would make since the password will still be stored in history even when passed to echo. ... but seems a little redundant/unnecessary, and perhaps gives a false sense of security.

True, I'm not sure that this enhances security much.
However, if we use `-p`, then we will have that ignore that docker warning every time.
This could negatively affect our security if we accidentally, due to habit, ignore a warning that _is_ important.

Reply:
I am not sure it's a security issue regardless, because what is stored in the history is the name of the variable, not the value itself. But I suppose this is an acceptable hack to work around the warning message.